### PR TITLE
[FIX] use the proper html template for mutli maps and label masker

### LINKED ISSUE #4299: [BUG] MultiNiftiMapsMasker reports are empty
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [X] Linux
- [ ] Mac
- [ ] Windows

### Operating system version

For example one of the following:
- Linux Ubuntu 22.04

### Python version

- [ ] 3.12
- [X] 3.11
- [ ] 3.10
- [ ] 3.9
- [ ] 3.8

### nilearn version

dev

### Expected behavior

to have non empty reports

### Current behavior & error messages

This is what I got:

image

image

### Steps and code to reproduce bug

```python
from pathlib import Path
from nilearn import datasets
from nilearn.maskers import (
    MultiNiftiMapsMasker
)

REPORTS_DIR = Path().cwd()

difumo = datasets.fetch_atlas_difumo(
    dimension=64, resolution_mm=2, legacy_format=False
)

data = datasets.fetch_development_fmri(n_subjects=2)

masker = MultiNiftiMapsMasker(
    maps_img=difumo.maps,
    standardize="zscore_sample",
    standardize_confounds="zscore_sample",
    memory="nilearn_cache",
    n_jobs=2,
)

masker.fit()
report = masker.generate_report()
report.save_as_html(REPORTS_DIR / "multi_nifti_maps_masker_atlas.html")

_ = masker.fit_transform(data.func, confounds=data.confounds)
report = masker.generate_report()
report.save_as_html(REPORTS_DIR / "multi_nifti_maps_masker_fitted.html")
```

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4299

Changes proposed in this pull request:

- use the appropriate HTML templates for reports of MultiNiftiMapsMasker and MultiNiftiLabelsMasker that would otherwise use the default one.

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Before: see https://github.com/nilearn/nilearn/issues/4299#issue-2161001560


After:
image

image

File: doc/changes/latest.rst [hunk 1]
```diff
@@ -11,6 +11,7 @@ Fixes
 - :bdg-dark:`Code` Fix plotting of carpet plot due to a change in the coming version of matplolib (3.9.0) (:gh:`4279` by `Rémi Gau`_).
 - :bdg-dark:`Code` Fix errant warning when using ``stat_type`` in :func:`nilearn.glm.compute_contrast` (:gh:`4257` by `Eric Larson`_).
 - :bdg-dark:`Code` Fix when thresholding is applied to images by GLM reports (:gh:`4258` by `Rémi Gau`_).
+- :bdg-dark:`Code` Use the appropriate HTML templates for :class:`nilearn.maskers.MultiNiftiMapsMasker` and :class:`nilearn.maskers.MultiNiftiLabelsMasker` reports that wowuld lead to empty reports for ``MultiNiftiMapsMasker`` (:gh:`4310` by `Rémi Gau`_).
```
Comment (bthirion) [inline doc/changes/latest.rst:14]:
```suggestion
- :bdg-dark:`Code` Use the appropriate HTML templates for :class:`nilearn.maskers.MultiNiftiMapsMasker` and :class:`nilearn.maskers.MultiNiftiLabelsMasker` reports that would lead to empty reports for ``MultiNiftiMapsMasker`` (:gh:`4310` by `Rémi Gau`_).
```

**Review by bthirion — COMMENTED** [review]
OK, so this PR is only about the template ?

**Comment by Remi-Gau** [comment]
> OK, so this PR is only about the template ?

yup, when investigating the root cause of #4299 turns out that report generation would not use the right template for the "multi" maskers and use the default one

Still not a 100% sure why this was only affecting the multimapsmasker and not also the multilabelsmasker, but I have checked the one generated for the latter and could not spot a difference before and after

we can wait till #4295 is in to do a proper check of all the maskers

**Comment by Remi-Gau** [comment]
# For review

Specifically compare and check multi label / maps maskers

### Before (current dev doc)

- maskers reports: https://nilearn.github.io/dev/modules/generated_reports/masker_reports_examples.html

### After

- maskers reports: https://output.circle-artifacts.com/output/job/288a1f19-4ea1-46f2-8425-6f0dc605d5f4/artifacts/0/dev/modules/generated_reports/masker_reports_examples.html

**Comment by Remi-Gau** [comment]
Actually this DOES fix a bug I had not noticed on the MultiNiftiLabelsMaskers:

The region summary button was not displayed.

Before

image

After

image

**Comment by Remi-Gau** [comment]
Messed up the rebase. The number of line changes makes no sense.
Fixing.

**Comment by Remi-Gau** [comment]
Note that the `next map` and `previous map` button do not seem to cycle through maps when the reports are embedded in the doc. But they work fine in jupyter notebooks.

**Review by bthirion — APPROVED** [review]
LGTM, thx !
